src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_SUBSTRING_START = 0;
+    private static final int DATE_SUBSTRING_END = 10;
+    private static final int HOURS_INDEX = 2;
+    private static final int RATE_INDEX = 3;
+    private static final String COLUMNS_SEPARATOR = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        //I split the getSalaryInfo method into 4 separate methods
+        int[] salary = salaryCounter(names, data, dateFrom, dateTo);
+        return resultExecutor(names, salary, dateFrom, dateTo);
+    }
+
+    public int[] salaryCounter(String[] names, String[] data, String dateFrom, String dateTo) {
+        int[] salary = new int[names.length];
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, DATE_FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, DATE_FORMATTER);
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                if (data[j].contains(names[i])) {
+                    LocalDate localDateCurrent = LocalDate.parse(data[j]
+                                    .substring(DATE_SUBSTRING_START,
+                                            DATE_SUBSTRING_END),
```

Comment:
I can see that you are doing a split by separator later in the code anyway, so I would recommend using it also here instead of the substrings because substrings are more of a hardcode approach and split is more agile because it can dynamically change depending on the input. But all of that is not crutical